java-symbol-solver-model/src/main/java/me/tomassetti/symbolsolver/model/typesystem/ReferenceTypeUsage.java

```diff
@@ -292,7 +292,8 @@ protected boolean compareConsideringTypeParameters(ReferenceTypeUsage other) {
         }
         if (this.getQualifiedName().equals(other.getQualifiedName())) {
             if (this.parameters().size() != other.parameters().size()) {
-                throw new IllegalStateException();
+                // one have the type not specified so they are compatible
+                return true;
```

Comment:
My hope is that once we have solid tests we can refactor mercilessly and be confident on the result. If we can resolve all the over 5000 method calls in the current JavaParser code (which uses streams and all the Java 8 stuff) we can be relatively sure that JavaSymbolSolver works.